markasread/background.js

```diff
@@ -1,130 +1,126 @@
-chrome.runtime.onInstalled.addListener(function(details) {
+chrome.runtime.onInstalled.addListener(async function(details) {
     if (details.reason == "update") {
-        chrome.storage.sync.get("visited", function(result) {
-            if (result["visited"] !== undefined) {
-                visited = result["visited"];
-                updateDictionary(visited);
-            }
-        })
+        const result = await chrome.storage.sync.get("visited")
+        if (result["visited"] !== undefined) {
+            const visited = result["visited"];
+            await updateDictionary(visited);
+        }
     }
     fetchMarkData();
 })
 
 function updateDictionary(visited) {
-    chrome.storage.local.set({ "visited": visited }, function() {
+    return chrome.storage.local.set({ "visited": visited }).then(() => {
```

Comment:
Why is `return` being used here? Do you have to use `return` when you call a function with await?

Reply:
I could have written this function in one of two ways, but the gist of it is that I just want the function to return a `Promise`. I chose to write it as a function that explicitly returns a promise, rather than as an `async function` that implicitly returns a `Promise`.

I could have also written it as an `async function` though, since `async function`s always wrap the return in a `Promise` (in this case `Promise<void>`):
```js
async function updateDictionary(visited) {
    await chrome.storage.local.set({ "visited": visited }).then(() => { 
        ... 
    })
}
```
But to answer your question, you use `await` on functions that return `Promise`s if you want to wait for the `Promise` to complete before moving to the next line of code.

So for a non-async function 

```js
function updateDictionary(visited){...}
```

you can only call it using `await` if `updateDictionary` returns a `Promise`.

```js
await updateDictionary(visited)
```

For an `async` function 

```js
async function updateDictionary(visited){...}
```

you can use `await` since `async function`s always wrap the return in a `Promise`, e.g., 

```js
async function someFunction() { 
   return 1 
}
``` 
returns a `Promise` that contains `1`.

Reply:
Thanks you, it's crystal clear.